src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java

```diff
@@ -464,8 +496,25 @@ else if (isInSameGroup(groupIdx, isStatic)) {
      * @return true if imports group are separated internally.
      */
     private boolean isSeparatorInGroup(int groupIdx, boolean isStatic, int line) {
-        final boolean inSameGroup = isInSameGroup(groupIdx, isStatic);
-        return (!separated || inSameGroup) && isSeparatorBeforeImport(line);
+        return shouldNoSeparatorBeforeImport(groupIdx, isStatic) && isSeparatorBeforeImport(line);
+    }
+
+    /**
+     * Checks whether there should be no separator before current import.
+     * @param groupIdx group number.
+     * @param isStatic whether the token is static or not.
+     * @return true if there should be no separator before current import.
+     */
+    private boolean shouldNoSeparatorBeforeImport(int groupIdx, boolean isStatic) {
```

Comment:
@romani 
As far as I remember. you were against negative words in the method names.

Reply:
done